modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetResponse.java

```diff
@@ -206,4 +219,26 @@ public NHttpClientConnection getConnection() {
     public ProtocolVersion getVersion() {
         return version;
     }
+
+    /**
+     * Retrieve the keep-alive timeout value
+     * @param keepAlive header value which has timeout in seconds
+     * @return keep-alive timeout in milliseconds
+     */
+    private int getKeepAliveTimeout(String keepAlive) {
+        int keepAliveTimeout = -1;
+        String timeout = keepAlive.split(KEEP_ALIVE_TIMEOUT)[1];
```

Comment:
What will happen if there is nothing after "timeout=" ? taking the [1] element will give an array index out of bound exception, isn't it?

Reply:
Fixed in https://github.com/wso2/wso2-synapse/pull/1472/commits/7bb5cad180112c3e018ae7a22ca6c67c4207ae78